src/main/java/org/ow2/proactive/catalog/util/LinkUtil.java

```diff
@@ -130,12 +131,51 @@ public static Link createRelativeLink(String bucketName, String objectName, Loca
     public static Link createRelativeLink(String bucketName, String objectName) {
         Link link = null;
         try {
-            link = new Link("buckets/" + bucketName + "/resources/" +
-                            URLEncoder.encode(objectName, "UTF-8")).withRel("relative");
+            link = new Link("buckets/" + bucketName + "/resources/" + encodeUrl(objectName)).withRel("relative");
 
         } catch (UnsupportedEncodingException e) {
             log.error("{} cannot be encoded", objectName, e);
         }
         return link;
     }
+
+    // ControllerLinkBuilder.linkTo has a problem: it double-URL-encodes characters!
+    // e.g. ' ' (whitespace) is encoded to '%2525' instead of '%20'
+    // TODO remove hack when https://github.com/spring-projects/spring-hateoas/issues/40 is resolved
+    private static final Field uriComponentsField;
+    static {
+        try {
+            uriComponentsField = LinkBuilderSupport.class.getDeclaredField("uriComponents");
+            uriComponentsField.setAccessible(true);
+        } catch (NoSuchFieldException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Method is created to avoid double encoding
+     * @param invocationValue
+     * @return encoded link
+     */
+    public static String linkTo(Object invocationValue) {
+        try {
+            final UriComponents uriComponents = (UriComponents) uriComponentsField.get(ControllerLinkBuilder.linkTo(invocationValue));
+            return uriComponents.toString();
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)